### PR TITLE
Get clang-tidy to work in CI

### PR DESCRIPTION
**Pull request by olanti-p**
#### Purpose of change
Get clang-tidy running in CI, fix most of the warnings

#### Describe the solution
* Install correct dependencies (ported from DDA)
* BN-specific fixes for the checks
* Silence text style lints in cata_libintl tests and misc lints in fmtlib (under assumption the code there is correct)
* Disable / fix other warnings

#### Testing
Github Actions

### CODE REVIEW

File: src/units_temperature.h [hunk 1]
```diff
@@ -82,6 +82,7 @@ inline constexpr quantity<value_type, temperature_in_millidegree_celsius_tag> fr
 {
     // Explicit casts to silence warnings about lossy conversions
     constexpr value_type max_temperature_fahrenheit = static_cast<value_type>( celsius_to_fahrenheit(
+                // NOLINTNEXTLINE(bugprone-integer-division): FIXME
                 static_cast<double>( std::numeric_limits<value_type>::max() / 1000 ) ) );
     const value_type temperature = v > max_temperature_fahrenheit ? max_temperature_fahrenheit : v;
     return from_millidegree_celsius<value_type>(
```
Comment (olanti-p) [inline src/units_temperature.h:88]: Is silencing it correct for this case?

The warning, for reference:
```
/home/runner/work/Cataclysm-BN/Cataclysm-BN/src/units_temperature.h:85:38: error: result of integer division used in a floating point context; possible loss of precision [bugprone-integer-division,-warnings-as-errors]
                static_cast<double>( std::numeric_limits<value_type>::max() / 1000 ) ) );
                                     ^
```

**Comment by olanti-p** [comment]
That should be it. Not sure what to do with the remaining 2 issues though:
1. Memory leak in `distribution_grid_tracker::make_distribution_grid_at()` (false positive?)
2. Unused statics check freaks out over expressions that end in macros. We have only one such case in BN:
https://github.com/cataclysmbnteam/Cataclysm-BN/blob/ddb7a8d89df51ea2e0eba3f4b6d6a06770722ee7/src/weather_gen.cpp#L21

**Comment by Coolthulhu** [comment]
> Unused statics check crashes on expressions that end in macros.

How about swapping the order to `M_PI * 2`? Would that help?

**Comment by olanti-p** [comment]
No idea. Maybe? I'm not familiar with clang-tidy internals at all.

**Comment by Coolthulhu** [comment]
> Memory leak in distribution_grid_tracker::make_distribution_grid_at() (false positive?)

Should be a false positive, but there may be a way to make it explicitly not a leak.
I think `clang-tidy` doesn't like the
```
return *dist_grid;
```
line itself, since `dist_grid` is going out of scope. It is not a real problem in current code, but if `overmap_positions` could be empty, this could lead to undefined behavior.
The simplest way could be to
```
return *parent_distribution_grids[submap_positions.front()];
```
instead of `dist_grid`.

**Comment by olanti-p** [comment]
Done and done, both suggestions worked.
The "memory leak" turned into some slightly different "memory leak", but it was easily to suppress this time.

**Comment by Coolthulhu** [comment]
Damn, that's a surprisingly high number of changes. Good work.

**Pull request merged**